plugins/modules/dcnm_inventory.py

```diff
@@ -674,7 +674,7 @@ def assign_role(self):
                     self.module.fail_json(msg=msg)
                 if role['ipAddress'] == create["switches"][0]['ipaddr']:
                     method = 'PUT'
-                    assign_path = '/fm/fmrest/topology/role/{}?newRole={}'.format(role['switchDbID'], create['role'])
+                    assign_path = '/fm/fmrest/topology/role/{}?newRole={}'.format(role['switchDbID'], create['role'].replace("_", "%20"))
```

Comment:
Were you able to verify that all role choices not work properly with this fix?

Reply:
The issue was with the role choices having "_" in there name(border_gateway, border_gateway_super_spine, etc). I have verified all the role choices with "_" in their name along with spine and border, and it works properly as expected.